File: Python/Majority Element.py
class Solution:
    # @param num, a list of integers
    # @return an integer
    def majorityElement(self, num):
        length = len(num)
        if length == 1 or length == 2:
            return num[0]
            
        bar = length / 2
        numberCount = {}
        for i in num:
            if i in numberCount:
                numberCount[i] = numberCount[i]+1
                if numberCount[i]>bar:
                    return i
            else:
                numberCount[i] = 1

File: Java/src/Majority Element.java
import java.util.HashMap;

public class Solution
{
  public int majorityElement(int[] num)
  {
    int length = num.length;
    if (length <= 2)
    {
      return num[0];
    }
    int bar = length / 2;

    Map<Integer, Integer> numberCount = new HashMap<>();
    for (int i : num)
    {

      if (numberCount.get(i) != null)
      {
        int count = numberCount.get(i) + 1;
        numberCount.put(i, count);
        if (count > bar)
        {
          return i;
        }
      }
      else
      {
        numberCount.put(i, 1);
      }
    }
    return 0;
  }
}
